lib/wallet/http.js

```diff
@@ -1084,6 +1084,36 @@ class HTTP extends Server {
       return res.json(200, mtx.getJSON(this.network));
     });
 
+    // Create auction-related transactions in advance (bid and reveal for now)
+    this.post('/wallet/:id/auction-txs', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      const bid = valid.u64('bid');
+      const lockup = valid.u64('lockup');
+      const passphrase = valid.str('passphrase');
+      const sign = valid.bool('sign', true);
+
+      assert(name, 'Name is required.');
+      assert(bid != null, 'Bid is required.');
+      assert(lockup != null, 'Lockup is required.');
+
+      const options = TransactionOptions.fromValidator(valid);
+      const {
```

Comment:
Nit: the codebase usually uses arrays when returning multiple things and destructuring them. I think returning an object is fine in this case since it could return more things in the future (additional transactions). I don't think there is a huge need to destructure out `bidMTX` and `revealMTX`, maybe something like this would be more in style:

```js
const auction = await req.wallet.createAuction(name, bid, lockup, options);

if (sign) {
  await req.wallet.sign(auction.bid, passphrase);
  await req.wallet.sign(auction.reveal, passphrase);
}

return res.json(200, {
  bid: auction.bid.getJSON(this.network),
  reveal: auction.reveal.getJSON(this.network)
});
```

Reply:
Alright, changed it !